### PR TITLE
[WIP] ansible-galaxy: Enable validate subcommand

### PR DESCRIPTION
**Pull request by Spredzy**
##### SUMMARY

Create a new validate subcommand to the ansible-galaxy binary that runs a
test suite with several checks:

  * ROLE001: Role has a meta/main.yml file
  * ROLE002: A LICENSE file is present in the role directory
  * ROLE003: A README.md file is present in the role directory
  * ROLE004: meta/main.yml contains basic necessary informations
  * ROLE005: Variables in defaults/ and vars/ are preceeded by role name
  * ROLE006: The dependencies listed in meta/main.yml are in Galaxy

There are some expectations about what an Ansible Role should look like at the filesystem level, this validate subcommand will check that they are respected.

##### ISSUE TYPE

* Feature Pull Request

##### COMPONENT NAME

* lib/ansible/galaxy/role.py

##### ANSIBLE VERSION

```
ansible 2.4.0 (mybranch 4cf3a3d78d) last updated 2017/05/05 23:13:49 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/spredzy/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/spredzy/Projects/github.com/Spredzy/ansible/lib/ansible
  executable location = /home/spredzy/Projects/github.com/Spredzy/ansible/venv/bin/ansible
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

##### ADDITIONAL INFORMATION

One of the main prupose of this feature is to be able to add `ansible-galaxy validate <path-to-role>` command in a CI pipeline. This would allow to gate every commit, and ensure commit remains "Galaxy Compliant".

* If everything is fine:

```
% ansible-galaxy validate ~/Projects/ansible-role/myansible-role 
% 
```

* if everything is fine (debug mode):

```
 % ansible-galaxy validate ~/Projects/ansible-role/myansible-role  -v
Using /etc/ansible/ansible.cfg as config file
[{'pass': True, 'optional': False, 'name': 'ROLE001', 'description': 'Role has a meta/mail.yml file'}, {'pass': True, 'optional': True, 'name': 'ROLE002', 'description': 'A LICENSE file is present in the role directory'}, {'pass': True, 'optional': False, 'name': 'ROLE003', 'description': 'A README.md file is present in the role directory'}, {'pass': True, 'optional': False, 'name': 'ROLE004', 'description': 'meta/main.yml contains basic necessary informations'}, {'pass': True, 'optional': True, 'name': 'ROLE005', 'description': u'Variables in defaults/ and vars/ are preceded by role name: openstack_certification'}, {'pass': True, 'optional': False, 'name': 'ROLE006', 'description': 'The dependencies listed in meta/main.yml are in Galaxy'}]
%
```

* Let's imagine there is no LICENSE file:

```
% ansible-galaxy validate ~/Projects/ansible-role/myansible-role 
[{'pass': False, 'optional': True, 'name': 'ROLE002', 'description': 'A LICENSE file is present in the role directory'}]
% 
```

### CODE REVIEW

**Comment by Spredzy** [comment]
This is a WIP, looking forward to engage conversation on what would be the best way to proceed while providing a workable example.

**Comment by gundalow** [comment]
Updated title to mark this as `[WIP]`, please update that once you are happy with the PR.

**Comment by gundalow** [comment]
Unit tests for Galaxy are failing. Please see https://docs.ansible.com/ansible/dev_guide/testing_units.html for info on how to run these.

File: lib/ansible/galaxy/role.py [hunk 1]
```diff
@@ -364,3 +366,103 @@ def spec(self):
...
+        """
+
+        results = []
+
+        # ROLE001
+        role001_result = {
+            'name': 'ROLE001',
+            'description': 'Role has a meta/mail.yml file',
+            'optional': False,
+            'pass': os.path.exists('%s/meta/main.yml' % self.path)
```
Comment (DanyC97) [inline lib/ansible/galaxy/role.py:382]: why does it need to be `meta/main.yml` ? shouldn't `y(a)ml` be supported? if not any chance you can point me where it says only `.yml` is supported ?

**Comment by gundalow** [comment]
This functionality will be in ansible-lint, most likely 4.1.0 https://github.com/ansible/ansible-lint/

**Pull request closed without merging**